File: LSP+ISP/Java/MarsRoverKata/src/main/java/MarsRoverKata/Mortar.java
package MarsRoverKata;

public class Mortar extends Projectile {
    public static final int Range = 10;

    public Mortar(Mars mars) {
        super(mars);
    }

    @Override
    protected int getMaxRange() {
        return Range;
    }

    @Override
    protected boolean isCollisionDetected(Point desired) {
        return false;
    }
}

File: DIP/C#_Exercise/MarsRoverKata/Mortar.cs
namespace MarsRoverKata
{
    public class Mortar : Projectile
    {
        public const int Range = 10;

        public Mortar(Mars mars)
            : base(mars)
        {
        }

        protected override int MaxRange
        {
            get { return Range; }
        }

        protected override bool IsCollisionDetected(Point desired)
        {
            return false;
        }
    }
}
